### PR TITLE
send queue: add state store backend for persisting pending events to disk

### PR DESCRIPTION
**Pull request by bnjbvr**
Ruma needs https://github.com/ruma/ruma/pull/1850. I need an Advil.

Part of #3361. The remaining bit will be to, at start, reload all the rooms' send queues that had _some_ pending event (right now, pending events will be automatically re-sent as soon as a timeline is opened for that room).

### CODE REVIEW

**Comment by bnjbvr** [comment]
I'm really not a big fan of the suggestions made by clippy here: https://github.com/matrix-org/matrix-rust-sdk/actions/runs/9601700440/job/26480909434?pr=3592#step:7:40

I imagine clippy's suggestion may avoid a few bounds check, but since this is testing code, why would we care?

File: crates/matrix-sdk-base/src/store/traits.rs [hunk 1]
```diff
@@ -369,27 +370,64 @@ pub trait StateStore: AsyncTraitDeps {
...
     ///
     /// * `room_id` - The `RoomId` of the room to delete.
     async fn remove_room(&self, room_id: &RoomId) -> Result<(), Self::Error>;
+
+    /// Save an event to be sent by a send queue later.
+    ///
+    /// # Arguments
+    ///
+    /// * `room_id` - The `RoomId` of the send queue's room.
+    /// * `transaction_id` - The unique key identifying the event to be sent
```
Comment (poljar) [inline crates/matrix-sdk-base/src/store/traits.rs:400]: Let us mention that the transaction ID must be random.

**Review by poljar — APPROVED** [review]
Left a small comment, I think this looks good.

**Comment by bnjbvr** [comment]
Thanks for the review meeting party :partying_face:

**Pull request merged**